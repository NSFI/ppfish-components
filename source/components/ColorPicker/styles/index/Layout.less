@prefixClassName: ~'@{prefixClass}-panel';

.@{prefixClassName}-inner {
  position: relative;
  border-radius: 4px;
  box-shadow: 0 1px 5px #ccc;
  border: 1px solid #ccc;
  padding-bottom: 8px;
}

.@{prefixClassName}-wrap {
  margin: 5px 0 0;
  height: 30px;
  width: 100%;
  position: relative;
  &-preview {
    position: absolute;
    right: 8px;
  }
  &-ribbon {
    position: absolute;
    left: 8px;
    top: 0;
    right: 43px;
    height: 30px;
  }
  &-alpha {
    position: absolute;
    left: 8px;
    right: 43px;
    bottom: 0;
    height: (30px - 5px) / 2;
  }
}

.@{prefixClassName}-wrap-has-alpha {
  .@{prefixClassName}-wrap-ribbon {
    height: (30px - 5px) / 2;
  }
}
