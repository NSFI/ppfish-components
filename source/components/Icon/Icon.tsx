import * as React from 'react';
import classNames from 'classnames';
import omit from 'omit.js';
import './Icon.less'


export interface IconProps {
  type: string;
  className?: string;
  title?: string;
  onClick?: React.MouseEventHandler<any>;
  spin?: boolean;
  style?: React.CSSProperties;
}

const Icon = (props: IconProps) => {
  const { type, className = '', spin } = props;
  const classString = classNames({
    fishdicon: true,
    'fishdicon-spin': !!spin || type === 'loading',
    [`fishdicon-${type}`]: true,
  }, className);
  return <i {...omit(props, ['type', 'spin'])} className={classString} />;
};

export default Icon;
