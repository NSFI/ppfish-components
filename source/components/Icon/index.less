@import "../../assets/css/lib/ant-theme-vars";
