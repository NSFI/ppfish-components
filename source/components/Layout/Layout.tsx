import * as React from 'react';
import classNames from 'classnames';

export interface GeneratorProps {
  prefixCls: string;
  tagName: 'header' | 'footer' | 'main' | 'section';
  displayName: string;
}
export interface BasicProps extends React.HTMLAttributes<HTMLDivElement> {
  prefixCls?: string;
  hasSider?: boolean;
}

export interface LayoutContextProps {
  siderHook: {
    addSider: (id: string) => void;
    removeSider: (id: string) => void;
  };
}
export const LayoutContext = React.createContext<LayoutContextProps>({
  siderHook: {
    addSider: () => null,
    removeSider: () => null,
  },
});

interface BasicPropsWithTagName extends BasicProps {
  tagName: 'header' | 'footer' | 'main' | 'section';
}

function generator({ prefixCls, tagName, displayName }: GeneratorProps) {
  return (BasicComponent: any) => {
    const Adapter: React.FC<BasicProps> = props => {
      return <BasicComponent prefixCls={prefixCls} tagName={tagName} {...props} />;
    };
    Adapter.displayName = displayName;
    return Adapter;
  };
}

const Basic = (props: BasicPropsWithTagName) => {
  const { prefixCls, className, children, tagName, ...others } = props;
  const classString = classNames(prefixCls, className);
  return React.createElement(tagName, { className: classString, ...others }, children);
};

const BasicLayout: React.FC<BasicPropsWithTagName> = props => {
  const [siders, setSiders] = React.useState<string[]>([]);

  const { prefixCls, className, children, hasSider, tagName: Tag, ...others } = props;
  const classString = classNames(
    prefixCls,
    {
      [`${prefixCls}-has-sider`]: typeof hasSider === 'boolean' ? hasSider : siders.length > 0,
    },
    className,
  );

  return (
    <LayoutContext.Provider
      value={{
        siderHook: {
          addSider: (id: string) => {
            setSiders(prev => [...prev, id]);
          },
          removeSider: (id: string) => {
            setSiders(prev => prev.filter(currentId => currentId !== id));
          },
        },
      }}
    >
      <Tag className={classString} {...others}>
        {children}
      </Tag>
    </LayoutContext.Provider>
  );
};

const Layout = generator({
  prefixCls: 'fishd-layout',
  tagName: 'section',
  displayName: 'Layout',
})(BasicLayout);

const Header = generator({
  prefixCls: 'fishd-layout-header',
  tagName: 'header',
  displayName: 'Header',
})(Basic);

const Footer = generator({
  prefixCls: 'fishd-layout-footer',
  tagName: 'footer',
  displayName: 'Footer',
})(Basic);

const Content = generator({
  prefixCls: 'fishd-layout-content',
  tagName: 'main',
  displayName: 'Content',
})(Basic);

export { Header, Footer, Content };

export default Layout;
