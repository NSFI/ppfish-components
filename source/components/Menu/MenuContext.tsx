import { createContext } from 'react';

export type MenuTheme = 'light' | 'dark';

export interface MenuContextProps {
  prefixCls: string;
  inlineCollapsed: boolean;
  fishdMenuTheme?: MenuTheme;
  firstLevel: boolean;
}

const MenuContext = createContext<MenuContextProps>({
  prefixCls: '',
  firstLevel: true,
  inlineCollapsed: false,
});

export default MenuContext;
