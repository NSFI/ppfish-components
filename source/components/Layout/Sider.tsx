// matchMedia polyfill for
// https://github.com/WickyNilliams/enquire.js/issues/82
if (typeof window !== 'undefined') {
  const matchMediaPolyfill = (mediaQuery: string): MediaQueryList => {
    return {
      media: mediaQuery,
      matches: false,
      addListener() {
      },
      removeListener() {
      },
    };
  };
  window.matchMedia = window.matchMedia || matchMediaPolyfill;
}

import * as React from 'react';
import * as PropTypes from 'prop-types';
import classNames from 'classnames';
import * as omit from 'omit.js';
import Icon from '../Icon/index';
import isNumeric from '../../utils/isNumeric';

const dimensionMap = {
  xs: '480px',
  sm: '576px',
  md: '768px',
  lg: '992px',
  xl: '1200px',
  xxl: '1600px',
};

export type CollapseType = 'clickTrigger' | 'responsive';

export type SiderTheme = 'light' | 'dark';

export interface SiderProps extends React.HTMLAttributes<HTMLDivElement> {
  prefixCls?: string;
  collapsible?: boolean;
  collapsed?: boolean;
  defaultCollapsed?: boolean;
  reverseArrow?: boolean;
  onCollapse?: (collapsed: boolean, type: CollapseType) => void;
  trigger?: React.ReactNode;
  width?: number | string;
  collapsedWidth?: number | string;
  breakpoint?: 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'xxl';
  theme?: SiderTheme;
  onBreakpoint?: (broken: boolean) => void;
}

export interface SiderState {
  collapsed?: boolean;
  below: boolean;
  belowShow?: boolean;
}

export interface SiderContext {
  siderCollapsed: boolean;
}

const generateId = (() => {
  let i = 0;
  return (prefix: string = '') => {
    i += 1;
    return `${prefix}${i}`;
  };
})();

export default class Sider extends React.Component<SiderProps, SiderState> {
  static __ANT_LAYOUT_SIDER: any = true;

  static defaultProps = {
    prefixCls: 'ant-layout-sider',
    collapsible: false,
    defaultCollapsed: false,
    reverseArrow: false,
    width: 200,
    collapsedWidth: 80,
    style: {},
    theme: 'dark' as SiderTheme,
  };

  static childContextTypes = {
    siderCollapsed: PropTypes.bool,
    collapsedWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  };

  static contextTypes = {
    siderHook: PropTypes.object,
  };

  private mql: MediaQueryList;
  private uniqueId: string;

  constructor(props: SiderProps) {
    super(props);
    this.uniqueId = generateId('ant-sider-');
    let matchMedia;
    if (typeof window !== 'undefined') {
      matchMedia = window.matchMedia;
    }
    if (matchMedia && props.breakpoint && props.breakpoint in dimensionMap) {
      this.mql = matchMedia(`(max-width: ${dimensionMap[props.breakpoint]})`);
    }
    let collapsed;
    if ('collapsed' in props) {
      collapsed = props.collapsed;
    } else {
      collapsed = props.defaultCollapsed;
    }
    this.state = {
      collapsed,
      below: false,
    };
  }

  getChildContext() {
    return {
      siderCollapsed: this.state.collapsed,
      collapsedWidth: this.props.collapsedWidth,
    };
  }

  componentWillReceiveProps(nextProps: SiderProps) {
    if ('collapsed' in nextProps) {
      this.setState({
        collapsed: nextProps.collapsed,
      });
    }
  }

  componentDidMount() {
    if (this.mql) {
      this.mql.addListener(this.responsiveHandler);
      this.responsiveHandler(this.mql);
    }

    if (this.context.siderHook) {
      this.context.siderHook.addSider(this.uniqueId);
    }
  }

  componentWillUnmount() {
    if (this.mql) {
      this.mql.removeListener(this.responsiveHandler);
    }

    if (this.context.siderHook) {
      this.context.siderHook.removeSider(this.uniqueId);
    }
  }

  responsiveHandler = (mql: MediaQueryList) => {
    this.setState({ below: mql.matches });
    const { onBreakpoint } = this.props;
    if (onBreakpoint) {
      onBreakpoint(mql.matches);
    }
    if (this.state.collapsed !== mql.matches) {
      this.setCollapsed(mql.matches, 'responsive');
    }
  }

  setCollapsed = (collapsed: boolean, type: CollapseType) => {
    if (!('collapsed' in this.props)) {
      this.setState({
        collapsed,
      });
    }
    const { onCollapse } = this.props;
    if (onCollapse) {
      onCollapse(collapsed, type);
    }
  }

  toggle = () => {
    const collapsed = !this.state.collapsed;
    this.setCollapsed(collapsed, 'clickTrigger');
  }

  belowShowChange = () => {
    this.setState({ belowShow: !this.state.belowShow });
  }

  render() {
    const { prefixCls, className, theme,
      collapsible, reverseArrow, trigger, style, width, collapsedWidth,
      ...others
    } = this.props;
    const divProps = omit(others, ['collapsed',
      'defaultCollapsed', 'onCollapse', 'breakpoint', 'onBreakpoint']);
    const rawWidth = this.state.collapsed ? collapsedWidth : width;
    // use "px" as fallback unit for width
    const siderWidth = isNumeric(rawWidth) ? `${rawWidth}px` : String(rawWidth);
    // special trigger when collapsedWidth == 0
    const zeroWidthTrigger = parseFloat(String(collapsedWidth || 0)) === 0 ? (
      <span onClick={this.toggle} className={`${prefixCls}-zero-width-trigger`}>
        <Icon type="bars" />
      </span>
    ) : null;
    const iconObj = {
      'expanded': reverseArrow ? <Icon type="right" /> : <Icon type="left" />,
      'collapsed': reverseArrow ? <Icon type="left" /> : <Icon type="right" />,
    };
    const status = this.state.collapsed ? 'collapsed' : 'expanded';
    const defaultTrigger = iconObj[status];
    const triggerDom = (
      trigger !== null ?
        zeroWidthTrigger || (
          <div className={`${prefixCls}-trigger`} onClick={this.toggle} style={{ width: siderWidth }}>
            {trigger || defaultTrigger}
          </div>
        ) : null
    );
    const divStyle = {
      ...style,
      flex: `0 0 ${siderWidth}`,
      maxWidth: siderWidth, // Fix width transition bug in IE11
      minWidth: siderWidth, // https://github.com/ant-design/ant-design/issues/6349
      width: siderWidth,
    };
    const siderCls = classNames(className, prefixCls, `${prefixCls}-${theme}`, {
      [`${prefixCls}-collapsed`]: !!this.state.collapsed,
      [`${prefixCls}-has-trigger`]: collapsible && trigger !== null && !zeroWidthTrigger,
      [`${prefixCls}-below`]: !!this.state.below,
      [`${prefixCls}-zero-width`]: parseFloat(siderWidth) === 0,
    });
    return (
      <div className={siderCls} {...divProps} style={divStyle}>
        <div className={`${prefixCls}-children`}>{this.props.children}</div>
        {collapsible || (this.state.below && zeroWidthTrigger) ? triggerDom : null}
      </div>
    );
  }
}
