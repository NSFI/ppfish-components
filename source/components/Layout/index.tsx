import Layout from './Layout';
import Sider from './Sider';

export { BasicProps as LayoutProps } from './Layout';
export { SiderProps } from './Sider';

Layout.Sider = Sider;
export default Layout;
