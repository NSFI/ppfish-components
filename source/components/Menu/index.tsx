import * as React from 'react';
import RcMenu, { ItemGroup, MenuProps as RcMenuProps } from './src';
import classNames from 'classnames';
import Icon from '../Icon';
import omit from 'rc-util/lib/omit';
import SubMenu, { SubMenuProps } from './SubMenu';
import Item, { MenuItemProps } from './MenuItem';
import warning from '../../utils/warning';
import { SiderContext, SiderContextProps } from '../Layout/Sider';
import collapseMotion from '../../utils/motion';
import { cloneElement } from '../../utils/reactNode';
import MenuContext, { MenuTheme } from './MenuContext';
import MenuDivider from './MenuDivider';

import './style/index.less';

export { MenuDividerProps } from './MenuDivider';

export { MenuItemGroupProps } from './src';

export type MenuMode = 'vertical' | 'vertical-left' | 'vertical-right' | 'horizontal' | 'inline';

export interface MenuProps extends RcMenuProps {
  theme?: MenuTheme;
  inlineIndent?: number;
}

type InternalMenuProps = MenuProps &
  SiderContextProps & {
    collapsedWidth?: string | number;
  };

class InternalMenu extends React.Component<InternalMenuProps> {
  static defaultProps: Partial<MenuProps> = {
    prefixCls: 'fishd-menu',
    // focusable: false,
    className: '',
    theme: 'light', // or dark
  };

  constructor(props: InternalMenuProps) {
    super(props);

    warning(
      !('inlineCollapsed' in props && props.mode !== 'inline'),
      '`inlineCollapsed` should only be used when `mode` is inline.',
    );

    warning(
      !(props.siderCollapsed !== undefined && 'inlineCollapsed' in props),
      '`inlineCollapsed` not control Menu under Sider. Should set `collapsed` on Sider instead.',
    );
  }

  getInlineCollapsed() {
    const { inlineCollapsed, siderCollapsed } = this.props;
    if (siderCollapsed !== undefined) {
      return siderCollapsed;
    }
    return inlineCollapsed;
  }

  renderMenu = () => {
    const { prefixCls, className, theme, expandIcon, ...restProps } = this.props;

    const passedProps = omit(restProps, ['siderCollapsed', 'collapsedWidth']);
    const inlineCollapsed = this.getInlineCollapsed();

    const defaultMotions = {
      horizontal: { motionName: `slide-up` },
      inline: collapseMotion,
      other: { motionName: `zoom-big` },
    };
    const menuClassName = classNames(`${prefixCls}-${theme}`, className);

    const direction = 'ltr';
    return (
      <MenuContext.Provider
        value={{
          prefixCls,
          inlineCollapsed: inlineCollapsed || false,
          fishdMenuTheme: theme,
          firstLevel: true,
        }}
      >
        <RcMenu
          // getPopupContainer={}
          overflowedIndicator={<Icon type="more-point" />}
          overflowedIndicatorPopupClassName={`${prefixCls}-${theme}`}
          {...passedProps}
          inlineCollapsed={inlineCollapsed}
          className={menuClassName}
          prefixCls={prefixCls}
          direction={direction}
          defaultMotions={defaultMotions}
          expandIcon={cloneElement(expandIcon, {
            className: `${prefixCls}-submenu-expand-icon`,
          })}
        />
      </MenuContext.Provider>
    );
  };

  render() {
    return this.renderMenu();
  }
}

// We should keep this as ref-able
class Menu extends React.Component<MenuProps, {}> {
  static Divider = MenuDivider;

  static Item = Item;

  static SubMenu = SubMenu;

  static ItemGroup = ItemGroup;

  render() {
    return (
      <SiderContext.Consumer>
        {(context: SiderContextProps) => <InternalMenu {...this.props} {...context} />}
      </SiderContext.Consumer>
    );
  }
}

export { MenuTheme, SubMenuProps, MenuItemProps };

export default Menu;
