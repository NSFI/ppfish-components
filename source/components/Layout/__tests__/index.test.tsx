import React, { useState } from 'react';
import { mount, render } from 'enzyme';
import Layout from '..';
import Icon from '../../Icon';
import Menu from '../../Menu';
import mountTest from '../../../../tools/tests/mountTest';

const { Sider, Content } = Layout;

describe('Layout', () => {
  mountTest(Layout);
  mountTest(Content);
  mountTest(Sider);
  mountTest(() => (
    <Layout>
      <Sider breakpoint="xs" />
      <Content />
    </Layout>
  ));

  it('detect the sider as children', () => {
    const wrapper = mount(
      <Layout>
        <Sider>Sider</Sider>
        <Content>Content</Content>
      </Layout>,
    );
    expect(wrapper.find('.fishd-layout').hasClass('fishd-layout-has-sider')).toBe(true);
    wrapper.unmount();
  });

  it('umount from multiple siders', async () => {
    const App = () => {
      const [hide1, setHide1] = useState(false);
      const [hide2, setHide2] = useState(false);
      return (
        <Layout>
          {hide1 ? null : <Sider>Sider</Sider>}
          {hide2 ? null : <Sider>Sider</Sider>}
          <Content>
            <button onClick={() => setHide1(true)} type="button">
              hide sider 1
            </button>
            <button onClick={() => setHide2(true)} type="button">
              hide sider 2
            </button>
          </Content>
        </Layout>
      );
    };
    const wrapper = mount(<App />);
    expect(wrapper.find('.fishd-layout').hasClass('fishd-layout-has-sider')).toBe(true);
    wrapper.find('button').at(0).simulate('click');
    expect(wrapper.find('.fishd-layout').hasClass('fishd-layout-has-sider')).toBe(true);
    wrapper.find('button').at(1).simulate('click');
    expect(wrapper.find('.fishd-layout').hasClass('fishd-layout-has-sider')).toBe(false);
  });

  it('detect the sider inside the children', async () => {
    const wrapper = mount(
      <Layout>
        <div>
          <Sider>Sider</Sider>
        </div>
        <Content>Content</Content>
      </Layout>,
    );
    expect(wrapper.find('.fishd-layout').hasClass('fishd-layout-has-sider')).toBe(true);
  });

  it('detect fishd-layout-sider-has-trigger class in sider when fishd-layout-sider-trigger div tag exists', async () => {
    const wrapper = mount(
      <Layout>
        <div>
          <Sider collapsible>Sider</Sider>
        </div>
        <Content>Content</Content>
      </Layout>,
    );
    expect(wrapper.find('.fishd-layout-sider').hasClass('fishd-layout-sider-has-trigger')).toBe(
      true,
    );
  });

  it('should have 50% width of sidebar', async () => {
    const wrapper = mount(
      <Layout>
        <div>
          <Sider width="50%">Sider</Sider>
        </div>
        <Content>Content</Content>
      </Layout>,
    );
    expect(wrapper.find('.fishd-layout-sider').at(0).prop('style').width).toBe('50%');
    expect(wrapper.find('.fishd-layout-sider').at(0).prop('style').flex).toBe('0 0 50%');
  });

  describe('zeroWidth', () => {
    it('detect fishd-layout-sider-zero-width class in sider when its width is 0%', async () => {
      const wrapper = mount(
        <Layout>
          <div>
            <Sider width="0%">Sider</Sider>
          </div>
          <Content>Content</Content>
        </Layout>,
      );
      expect(wrapper.find('.fishd-layout-sider').hasClass('fishd-layout-sider-zero-width')).toBe(
        true,
      );
    });

    describe('should collapsible', () => {
      it('uncontrolled', () => {
        const onCollapse = jest.fn();

        const wrapper = mount(
          <Layout>
            <Sider collapsible breakpoint="lg" collapsedWidth="0" onCollapse={onCollapse}>
              Sider
            </Sider>
            <Content>Content</Content>
          </Layout>,
        );

        onCollapse.mockReset();

        wrapper.find('.fishd-layout-sider-zero-width-trigger').simulate('click');
        expect(onCollapse).toHaveBeenCalledTimes(1);
      });

      it('controlled', () => {
        const Demo = () => {
          const [collapsed, setCollapsed] = React.useState(true);

          return (
            <Layout>
              <Sider
                collapsed={collapsed}
                collapsible
                breakpoint="lg"
                collapsedWidth="0"
                onCollapse={setCollapsed}
              >
                Sider
              </Sider>
              <Content>Content</Content>
            </Layout>
          );
        };

        const wrapper = mount(<Demo />);
        expect(wrapper.find(Sider).prop('collapsed')).toBeTruthy();

        wrapper.find('.fishd-layout-sider-zero-width-trigger').simulate('click');
        expect(wrapper.find(Sider).prop('collapsed')).toBeFalsy();
      });
    });
  });

  it('detect fishd-layout-sider-dark as default theme', async () => {
    const wrapper = mount(<Sider>Sider</Sider>);
    expect(wrapper.find('.fishd-layout-sider').hasClass('fishd-layout-sider-dark')).toBe(true);
  });

  it('detect fishd-layout-sider-light when set light theme', async () => {
    const wrapper = mount(<Sider theme="light">Sider</Sider>);
    expect(wrapper.find('.fishd-layout-sider').hasClass('fishd-layout-sider-light')).toBe(true);
  });

  it('renders string width correctly', () => {
    const wrapper = render(<Sider width="200">Sider</Sider>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should be controlled by collapsed', () => {
    const wrapper = mount(<Sider>Sider</Sider>);
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({ collapsed: true });
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it('should not add fishd-layout-has-sider when `hasSider` is `false`', () => {
    const wrapper = mount(
      <Layout hasSider={false}>
        <Sider>Sider</Sider>
      </Layout>,
    );
    expect(wrapper.find('.fishd-layout').hasClass('fishd-layout-has-sider')).toBe(false);
  });

  it('render correct with Tooltip', () => {
    jest.useFakeTimers();
    const wrapper = mount(
      <Sider collapsible collapsed={false}>
        <Menu mode="inline">
          <Menu.Item key="1">
            <Icon type="user" />
            <span>Light</span>
          </Menu.Item>
        </Menu>
      </Sider>,
    );

    wrapper.find('.fishd-menu-item').hostNodes().simulate('mouseenter');
    jest.runAllTimers();
    wrapper.update();
    expect(wrapper.find('.fishd-tooltip-inner').length).toBeFalsy();
    wrapper.find('.fishd-menu-item').hostNodes().simulate('mouseout');
    jest.runAllTimers();
    wrapper.update();

    wrapper.setProps({ collapsed: true });
    wrapper.find('.fishd-menu-item').hostNodes().simulate('mouseenter');
    jest.runAllTimers();
    wrapper.update();
    expect(wrapper.find('.fishd-tooltip-inner').length).toBeTruthy();

    jest.useRealTimers();
  });
});

describe('Sider', () => {
  const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});

  afterEach(() => {
    errorSpy.mockReset();
  });

  afterAll(() => {
    errorSpy.mockRestore();
  });

  // it('should trigger onBreakpoint', async () => {
  //   jest.useFakeTimers();
  //   const onBreakpoint = jest.fn(e => {
  //   });

  //   mount(
  //     <Sider
  //       breakpoint="md"
  //       collapsedWidth="10"
  //       onBreakpoint={onBreakpoint}
  //     >
  //       Sider
  //     </Sider>,
  //   );

  //   jest.runAllTimers();
  //   expect(onBreakpoint).toHaveBeenCalledWith(true);
  // });

  it('should warning if use `inlineCollapsed` with menu', () => {
    mount(
      <Sider collapsible>
        <Menu mode="inline" inlineCollapsed />
      </Sider>,
    );
    expect(errorSpy).toHaveBeenCalledWith(
      'Warning: `inlineCollapsed` not control Menu under Sider. Should set `collapsed` on Sider instead.',
    );
  });

  it('zeroWidthTriggerStyle should work', () => {
    const wrapper = mount(
      <Sider collapsedWidth={0} collapsible zeroWidthTriggerStyle={{ background: '#F96' }}>
        <Menu theme="dark" mode="inline" defaultSelectedKeys={['1']}>
          <Menu.Item key="1">
            <Icon type="user" />
            <span>nav 1</span>
          </Menu.Item>
        </Menu>
      </Sider>,
    );
    expect(wrapper.find('.fishd-layout-sider-zero-width-trigger').props().style).toEqual({
      background: '#F96',
    });
  });

  it('should be able to customize zero width trigger by trigger prop', () => {
    const wrapper = mount(
      <Sider collapsedWidth={0} collapsible trigger={<span className="my-trigger" />}>
        <Menu theme="dark" mode="inline" defaultSelectedKeys={['1']}>
          <Menu.Item key="1">
            <Icon type="user" />
            <span>nav 1</span>
          </Menu.Item>
        </Menu>
      </Sider>,
    );
    expect(wrapper.find('.fishd-layout-sider-zero-width-trigger').find('.my-trigger').length).toBe(
      1,
    );
  });

  it('should get aside element from ref', () => {
    const ref = React.createRef<HTMLDivElement>();
    const onSelect = jest.fn();

    mount(
      <Sider onSelect={onSelect} ref={ref}>
        Sider
      </Sider>,
    );
    expect(ref.current instanceof HTMLElement).toBe(true);
  });
});
