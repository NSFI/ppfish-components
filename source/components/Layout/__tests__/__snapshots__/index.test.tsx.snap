// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders string width correctly 1`] = `
<aside
  class="fishd-layout-sider fishd-layout-sider-dark"
  style="flex:0 0 200px;max-width:200px;min-width:200px;width:200px"
>
  <div
    class="fishd-layout-sider-children"
  >
    Sider
  </div>
</aside>
`;

exports[`Layout should be controlled by collapsed 1`] = `
<Sider
  collapsedWidth={64}
  collapsible={false}
  defaultCollapsed={false}
  prefixCls="fishd-layout-sider"
  reverseArrow={false}
  style={Object {}}
  theme="dark"
  width={160}
>
  <aside
    className="fishd-layout-sider fishd-layout-sider-dark"
    style={
      Object {
        "flex": "0 0 160px",
        "maxWidth": "160px",
        "minWidth": "160px",
        "width": "160px",
      }
    }
  >
    <div
      className="fishd-layout-sider-children"
    >
      Sider
    </div>
  </aside>
</Sider>
`;

exports[`Layout should be controlled by collapsed 2`] = `
<Sider
  collapsed={true}
  collapsedWidth={64}
  collapsible={false}
  defaultCollapsed={false}
  prefixCls="fishd-layout-sider"
  reverseArrow={false}
  style={Object {}}
  theme="dark"
  width={160}
>
  <aside
    className="fishd-layout-sider fishd-layout-sider-dark fishd-layout-sider-collapsed"
    style={
      Object {
        "flex": "0 0 64px",
        "maxWidth": "64px",
        "minWidth": "64px",
        "width": "64px",
      }
    }
  >
    <div
      className="fishd-layout-sider-children"
    >
      Sider
    </div>
  </aside>
</Sider>
`;
